File: testdata/code/cpp/simple.cpp
#include <string>
#include <vector>
#include <memory>

const int MAX_CONNECTIONS = 100;
const std::string DEFAULT_HOST = "localhost";

static int global_counter = 0;

struct Point {
    double x;
    double y;

    Point(double x, double y) : x(x), y(y) {}
};

class User {
private:
    std::string id;
    std::string name;
    std::string email;

public:
    User(const std::string& id, const std::string& name, const std::string& email)
        : id(id), name(name), email(email) {}

    std::string getId() const { return id; }
    std::string getName() const { return name; }
    std::string getEmail() const { return email; }

    bool validate() const {
        return email.find('@') != std::string::npos;
    }
};

template<typename T>
class Repository {
private:
    std::vector<T> items;

public:
    Repository() {}

    void add(const T& item) {
        items.push_back(item);
    }

    size_t size() const {
        return items.size();
    }

    const T* get(size_t index) const {
        if (index < items.size()) {
            return &items[index];
        }
        return nullptr;
    }
};

typedef Repository<User> UserRepository;

std::unique_ptr<User> createUser(const std::string& id, const std::string& name, const std::string& email) {
    return std::make_unique<User>(id, name, email);
}

File: testdata/code/rust/simple.rs
use std::collections::HashMap;
use std::fmt;

const MAX_USERS: usize = 1000;
const DEFAULT_TIMEOUT: u64 = 30;

static mut GLOBAL_COUNTER: i32 = 0;

#[derive(Debug, Clone)]
pub struct User {
    pub id: String,
    pub name: String,
    pub email: String,
}

impl User {
    pub fn new(id: String, name: String, email: String) -> Self {
        User { id, name, email }
    }

    pub fn validate(&self) -> bool {
        self.email.contains('@')
    }
}

pub trait Repository<T> {
    fn add(&mut self, item: T) -> Result<(), String>;
    fn get(&self, id: &str) -> Option<&T>;
    fn remove(&mut self, id: &str) -> Option<T>;
}

pub struct UserRepository {
    users: HashMap<String, User>,
}

impl UserRepository {
    pub fn new() -> Self {
        UserRepository {
            users: HashMap::new(),
        }
    }

    pub fn size(&self) -> usize {
        self.users.len()
    }
}

impl Repository<User> for UserRepository {
    fn add(&mut self, user: User) -> Result<(), String> {
        if self.users.len() >= MAX_USERS {
            return Err("Repository full".to_string());
        }
        self.users.insert(user.id.clone(), user);
        Ok(())
    }

    fn get(&self, id: &str) -> Option<&User> {
        self.users.get(id)
    }

    fn remove(&mut self, id: &str) -> Option<User> {
        self.users.remove(id)
    }
}

pub fn create_user(id: &str, name: &str, email: &str) -> User {
    User::new(id.to_string(), name.to_string(), email.to_string())
}

pub enum Status {
    Active,
    Inactive,
    Pending,
}
